mm/mm_gran/mm_grantable.c

```diff
@@ -223,17 +223,47 @@ bool gran_match(const gran_t *gran, size_t posi, size_t size, bool used,
 
   if (mpos && !used)
     {
-      /* offset of last used when matching for free */
+      size_t tmp;
 
+      v = gran->gat[c];
```

Comment:
could we use function from nuttx/include/nuttx/bits.h, which already do many optimizations?

Reply:
You mean to replace the whole search / match algorithm with find_next_bit / find_next_zero_bit? I'd rather not do that, @yf13 can comment if that's feasible.

I only noticed that finding a free granule takes longer the more the memory is filled up.

The builtin ctz/clz have a time complexity of O(log(n))